clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -749,7 +749,7 @@ const llvm::fltSemantics &FP80Type::getFloatSemantics() const {
 llvm::TypeSize
 FP80Type::getTypeSizeInBits(const mlir::DataLayout &dataLayout,
                             mlir::DataLayoutEntryListRef params) const {
-  return llvm::TypeSize::getFixed(16);
+  return llvm::TypeSize::getFixed(80);
```

Comment:
The size of `FP80Type` should be 16 bytes, or 128 bits. The lower 80 bits are used and the higher 48 bits are just paddings.

```suggestion
  return llvm::TypeSize::getFixed(128);
```